diff -uprN a/arch/mips/Kconfig b/arch/mips/Kconfig
--- a/arch/mips/Kconfig	2025-06-11 20:29:25.728329642 +0900
+++ b/arch/mips/Kconfig	2025-06-11 20:35:12.165402040 +0900
@@ -1002,6 +1002,15 @@ config RTL931X
 	select SYS_SUPPORTS_SMP
 	select SYS_SUPPORTS_MIPS_CPS
 
+config RTL8198D
+	bool "Realtek RTL9198D based platforms"
+	depends on MACH_REALTEK_RTL
+	select CPU_SUPPORTS_CPUFREQ
+	select MIPS_CPU_SCACHE
+	select MIPS_EXTERNAL_TIMER
+	select SYS_SUPPORTS_MULTITHREADING
+
 source "arch/mips/alchemy/Kconfig"
 source "arch/mips/ath25/Kconfig"
 source "arch/mips/ath79/Kconfig"
